collagraph/__init__.py

```diff
@@ -439,7 +516,30 @@ def is_equivalent_event_handler(val, other, key):
                 continue
 
             event_type = key_to_event(name)
-            self.renderer.add_event_listener(dom, event_type, next_props[name])
+            events_to_add[event_type] = next_props[name]
+
+        if fiber.parent and fiber.parent.component:
+            if events_to_remove or events_to_add or attrs_to_remove or attrs_to_update:
+                if fiber.parent.component._mounted:
+                    fiber.parent.component.before_update()
```

Comment:
This feels very brittle. I don't have a better solution yet though.

Reply:
depending on how critical some of these lifecycle hooks are, you could consider just leaving them out

Reply:
Marking this as resolved.